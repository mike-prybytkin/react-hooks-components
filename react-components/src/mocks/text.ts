const mockText = {
  homeLink: 'home',
  aboutLink: 'about',
  cardTitle: 'Fjallraven',
  searchPlaceholder: 'Search...',
  linkBackToHome: 'Back to home page',
  heading404: 'Error 404!',
  headingAboutUs: 'About Us!',
  headingMain: 'Welcome to us!',
  pageNotReady: 'Sorry, page not ready yet...',
  loading: 'Loading ...',
  itemNotFound: 'Sorry, item not found...',
  headingForm: 'Create User!',
  notCreatedUserInForm: 'Sorry, no users have been created yet...',
  labelUserName: 'Name:',
  placeholderUserName: 'Your Name',
  labelUserBirthday: 'Birthday:',
  labelUserSalary: 'Your monthly income:',
  switcherGender: 'Gender:',
  switcherGenderMale: 'Male',
  switcherGenderFemale: 'Female',
  labelUserMailing: 'Receive mailing',
  labelFileUpload: 'Add your photo',
  createUserButton: 'Create user',
  productCardButton: 'Learn more',
  prevButton: 'prev',
  nextButton: 'next',
  userSex: 'sex:',
  userBDay: 'b-day:',
  userSalary: 'salary:',
  subscribed: 'Subscribed',
  unsubscribed: 'Unsubscribed',
  toMailings: 'to mailings!',
  fromMailings: 'from mailings...',
};

export default mockText;
