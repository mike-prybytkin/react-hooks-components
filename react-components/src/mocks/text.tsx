const mockText = {
  homeLink: 'home',
  aboutLink: 'about',
  cardTitle: 'Fjallraven',
  searchPlaceholder: 'Search...',
  linkBackToHome: 'Back to home page',
  heading404: 'Error 404!',
  headingAboutUs: 'About Us!',
  headingMain: 'Welcome to our store!',
  pageNotReady: 'Sorry, page not ready yet...',
  loading: 'Loading ...',
  itemNotFound: 'Sorry, item not found...',
  headingForm: 'Create User!',
  notCreatedUserInForm: 'Sorry, no users have been created yet...',
  labelUserName: 'Name:',
  placeholderUserName: 'Your Name',
  labelUserBirthday: 'Birthday:',
  labelUserSalary: 'Your monthly income:',
  switcherGender: 'Gender:',
  switcherGenderMale: 'Male',
  switcherGenderFemale: 'Female',
  labelUserMailing: 'Receive mailing',
  labelFileUpload: 'Add your photo',
  createUserButton: 'Create user',
};

export default mockText;
