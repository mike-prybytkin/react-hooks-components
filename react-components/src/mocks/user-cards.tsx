import { IUserCard } from 'share/types';
import { userPhoto } from './user-photo';

export const mockUserCards: IUserCard[] = [
  {
    name: 'Max',
    avatarPath: userPhoto.max,
    gender: 'Male',
    birthday: '12-08-1995',
    salary: 'less than 100$',
    mailing: true,
  },
  {
    name: 'Kate',
    avatarPath: userPhoto.kate,
    gender: 'Female',
    birthday: '01-12-2003',
    salary: '200$ - 299$',
    mailing: false,
  },
];
