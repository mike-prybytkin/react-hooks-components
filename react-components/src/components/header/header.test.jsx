import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Header from './header';
import { mockCards } from '../../mocks/cards';
import mockText from '../../mocks/text';

const setUp = () =>
  render(<Header cards={mockCards} onSearch={mockCards} />, { wrapper: BrowserRouter });

describe('Header component', () => {
  it('should render links container', () => {
    setUp();
    expect(screen.queryByRole('list')).toBeInTheDocument();
  });

  it('should render link to About page', () => {
    setUp();
    const linkAbout = screen.getByText(new RegExp(mockText.aboutLink, 'i'));
    expect(linkAbout).toBeInTheDocument();
  });

  it('should create snapshot', () => {
    const { asFragment } = setUp();
    expect(asFragment()).toMatchSnapshot();
  });
});
