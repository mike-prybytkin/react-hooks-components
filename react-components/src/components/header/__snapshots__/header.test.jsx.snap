// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should create snapshot 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <nav
      class="navbar"
    >
      <ul
        class="navbar__items"
      >
        <li
          class="navbar__item"
        >
          <a
            aria-current="page"
            class="navbar__link active"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="navbar__item"
        >
          <a
            class="navbar__link"
            href="/about"
          >
            About
          </a>
        </li>
        <li
          class="navbar__item"
        >
          <a
            class="navbar__link"
            href="/form"
          >
            Form
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="search-container"
    >
      <form
        class="search-form"
      >
        <span>
          üîç
        </span>
        <input
          class="search"
          placeholder="Search..."
          value=""
        />
      </form>
    </div>
  </header>
</DocumentFragment>
`;
